### PR TITLE
auth: Add some UeberBackend unit tests, document multi-backend interactions

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

This PR adds a few unit tests for the `UeberBackend` and tries to document a few corner case of multi-backend interactions.
Needs a thorough review by someone that understands how PowerDNS is supposed to work with multiple backends :-) 

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [x] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
Any feedback on this pull request? :-)

File: docs/appendices/backend-writers-guide.rst [hunk 1]
```diff
@@ -771,6 +813,147 @@ other update/remove functionality at a later stage.
   must be added to the zone. ``rrset`` can be empty in which case the
   method is used to remove a RRset.
 
+Domain metadata support
+-----------------------
+
+As described in :ref:`per-zone-settings-domain-metadata`, each served zone can have “metadata”. Such metadata determines how this zone behaves in certain circumstances.
```
Comment (zeha) [inline docs/appendices/backend-writers-guide.rst:819]: unsure if the quoting around metadata is the correct style.

**Comment by zeha** [comment]
Text/docs make sense to me. Didn't really look at the test cases, but if they pass... :)

👍 ❤️

**Comment by zeha** [comment]
> Didn't really look at the test cases, but if they pass... :)

Spent some more time on the tests, lgtm. The "overlapping" tests would need changing after #9007 I guess.

**Comment by Habbie** [comment]
A rebase should make the spelling notices go away.

**Comment by rgacogne** [comment]
Rebased!

> The "overlapping" tests would need changing after #9007 I guess.

Indeed, #9007 needs a bit more work anyway so I'll fix it there.

File: docs/appendices/backend-writers-guide.rst (hunk 1 above)
Comment (rgacogne) [reply docs/appendices/backend-writers-guide.rst:819]: The idea was to link to https://docs.powerdns.com/authoritative/domainmetadata.html#per-zone-settings-domain-metadata but I'm afraid it's not working, I'll have a look.
